net/dns/resolver/forwarder.go

```diff
@@ -183,6 +198,13 @@ func clampEDNSSize(packet []byte, maxSize uint16) {
 	binary.BigEndian.PutUint16(opt[3:5], maxSize)
 }
 
+// getEDNSBufferSize extracts the EDNS buffer size from a DNS request packet.
+// Returns 0 if no EDNS OPT record is found or if there's an error.
+func getEDNSBufferSize(packet []byte) uint16 {
```

Comment:
Rather than returning 0, it would be more idiomatic to return `(_ uint16, ok bool)` so a missing record is explicitly communicated rather than implied with a size of 0.